packages/components/src/popover/index.js

```diff
@@ -66,31 +68,33 @@ const positionToPlacement = ( position ) => {
 	return y;
 };
 
+// Given the placement, compute the animation origin used for the entrance
+// animation. The origin should the the "opposite" of the placement.
 const placementToAnimationOrigin = ( placement ) => {
-	const [ a, b ] = placement.split( '-' );
-
-	let x, y;
-	if ( a === 'top' || a === 'bottom' ) {
-		x = a === 'top' ? 'bottom' : 'top';
-		y = 'middle';
-		if ( b === 'start' ) {
-			y = 'left';
-		} else if ( b === 'end' ) {
-			y = 'right';
+	const [ placementX, placementY ] = placement.split( '-' );
```

Comment:
> But I think it's more succinct and easier to read to have a map instead of this function

Good point, done in https://github.com/WordPress/gutenberg/pull/42531/commits/3b7f86ccb8605fba7f253c3be4b3ebbbe70d7d62

> not sure if I got all the values right

Since these values are passed to `getAnimateClassName`, I had a closer look at the values that are accepted for the `appear` animation:
https://github.com/WordPress/gutenberg/blob/b2208814a49c12ad206396502fb72ac91af52f4d/packages/components/src/animate/index.js#L7

This lead me to make a few tweaks

- replaced `center` with `middle`
- replaced `top middle` with `top` and `bottom middle` with `bottom`

I also noticed how the previous function was assigning `left` to `start` and `right` to `end`, which goes against our understanding of what the "opposite side" term means.

> I also wonder whether the animation origin should take RTL languages into account. Though maybe that's flipped elsewhere.

I don't think it is, because floating-ui already uses logical positioning (end and start), and so it should be the popover's task to convert them to left or right based on the pages' writing direction.

I added RTL support tentatively in https://github.com/WordPress/gutenberg/pull/42531/commits/c8a820b5250b9d00791f540094c0b67d7b02b0a6

Reply:
Actually, since [snapshots unit tests are failing](https://github.com/WordPress/gutenberg/runs/7447791421?check_suite_focus=true), and looking [at the snapshots](https://github.com/WordPress/gutenberg/pull/40740/files#diff-d825db71d6c16d0d5f60077604c1fc68de77e65af0ef4d10074f9d6d8f646a51) prior to Riad's refactor, I decided to update the mapping so that it produces the same output as the function that it replaces https://github.com/WordPress/gutenberg/pull/42531/commits/7c0339cc2dd72b1d3aaa4d97281804d4b9189e91

Reply:
If we spot any issues it'll be an easy fix, so sounds good 👍